charmcraft/parts.py

```diff
@@ -149,13 +151,32 @@ def get_build_snaps(cls) -> Set[str]:
 
     def get_build_packages(self) -> Set[str]:
         """Return a set of required packages to install in the build environment."""
-        return {
-            "python3-pip",
-            "python3-setuptools",
-            "python3-wheel",
-            "python3-venv",
-            "python3-dev",
-        }
+        if platform.is_deb_based():
+            return {
+                "python3-dev",
+                "python3-pip",
+                "python3-setuptools",
+                "python3-venv",
+                "python3-wheel",
+            }
+        elif platform.is_yum_based():
+            if os_utils.OsRelease().version_id() == "7":
```

Comment:
This works because only CentOS is listed as yum-based. A more robust way to do this verification would be to check both the distribution name and version here (just to be sure).

Reply:
Also `is_yum_based` is in an unreleased version of craft-parts, maybe we should land this PR in a separate branch that uses craft parts from main? (some fixes are still necessary before relasing stable 1.19.0)

Reply:
Or we can wait until the features are available in a point release.